### PR TITLE
test(e2e): add trailing newline edge case for batch input

### PR DESCRIPTION
**Pull request by yizhaodev**
## What does this PR do?

E2E test verifying input JSONL files with trailing newlines (one or two) produce correct request counts and don't cause parse errors.

## How was this tested?

- [x] Unit tests added/updated/verified
- [x] Integration/e2e tests added/updated/verified
- [x] Manual testing performed

## Checklist

- [x] Commits are signed off (`git commit -s`) per [DCO](PR_SIGNOFF.md)
- [x] Code follows project [contributing guidelines](CONTRIBUTING.md)
- [x] CI checks pass (`make ci`)
- [ ] E2E tests pass (`make test-e2e`)

## Related Issues

None. Standalone correctness fix.

### CODE REVIEW

File: test/e2e/batches_test.go [hunk 1]
```diff
@@ -303,6 +304,45 @@ func doTestBatchSharedInputFile(t *testing.T) {
 	}
 }
 
+// doTestBatchTrailingNewline verifies that trailing newlines in input files
```
Comment (j-mok-dev) [inline test/e2e/batches_test.go]: this test is asserting behavior that the current implementation does not provide yet. `preProcessJob()` still passes blank lines to `extractAndValidateLine()`, so an input ending with `\n\n` becomes an empty JSON record and should fail ingestion rather than complete with `total=2`. Could you either include the parser-side fix in this PR or narrow the test to the currently supported case?

Comment (yizhaodev) [reply test/e2e/batches_test.go]: good point, changed to test with only 1 trailing newline
let's handle more edge cases in the future

**Review by lioraron — APPROVED** [review]
